src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp

```diff
@@ -606,121 +606,99 @@ class boss_thaddius_summon : public CreatureScript
     };
 };
 
-class spell_thaddius_pos_neg_charge : public SpellScriptLoader
+class spell_thaddius_pos_neg_charge : public SpellScript
 {
-public:
-    spell_thaddius_pos_neg_charge() : SpellScriptLoader("spell_thaddius_pos_neg_charge") { }
+    PrepareSpellScript(spell_thaddius_pos_neg_charge);
 
-    class spell_thaddius_pos_neg_charge_SpellScript : public SpellScript
+    void HandleTargets(std::list<WorldObject*>& targets)
     {
-        PrepareSpellScript(spell_thaddius_pos_neg_charge_SpellScript);
-
-        void HandleTargets(std::list<WorldObject*>& targets)
+        uint8 count = 0;
+        for (auto& ihit : targets)
         {
-            uint8 count = 0;
-            for (auto& ihit : targets)
+            if (ihit->GetGUID() != GetCaster()->GetGUID())
             {
-                if (ihit->GetGUID() != GetCaster()->GetGUID())
+                if (Player* target = ihit->ToPlayer())
                 {
-                    if (Player* target = ihit->ToPlayer())
+                    if (target->HasAura(GetTriggeringSpell()->Id))
                     {
-                        if (target->HasAura(GetTriggeringSpell()->Id))
-                        {
-                            ++count;
-                        }
+                        ++count;
                     }
                 }
             }
-
-            if (count)
-            {
-                uint32 spellId = GetSpellInfo()->Id == SPELL_POSITIVE_CHARGE ? SPELL_POSITIVE_CHARGE_STACK : SPELL_NEGATIVE_CHARGE_STACK;
-                GetCaster()->SetAuraStack(spellId, GetCaster(), count);
-            }
         }
 
-        void HandleDamage(SpellEffIndex /*effIndex*/)
+        if (count)
         {
-            if (!GetTriggeringSpell())
-                return;
+            uint32 spellId = GetSpellInfo()->Id == SPELL_POSITIVE_CHARGE ? SPELL_POSITIVE_CHARGE_STACK : SPELL_NEGATIVE_CHARGE_STACK;
```

Comment:
need to validate :)

Reply:
Even if the spell is not used/casted? It's just an enum

Reply:
it is used
```cpp
GetCaster()->SetAuraStack(spellId, GetCaster(), count);
```